src/main/java/com/aws/greengrass/smbridge/MessageBridge.java

```
@@ -99,11 +108,26 @@ private void handleMessage(MQTTMessage message) {
                     LOGGER.atError().setCause(e).kv("Stream", stream).log("Stream Publish failed");
                 }
             };
-            // Perform topic matching on filter
+            // Perform topic matching on filter from mapped topics/destinations
             sourceDestinationMap.entrySet().stream().filter(
                     mapping -> MqttTopic.isMatched(mapping.getKey(), sourceTopic))
                     .map(Map.Entry::getValue)
                     .forEach(perTopicMapping -> perTopicMapping.forEach(processDestination));
+
+            // TODO: Handle the case where reserved topic is already matched by a user-configured mapping
+            // Perform topic matching against reserved topic
+            if (MqttTopic.isMatched(SMBridge.RESERVED_TOPIC, sourceTopic)) {
+                String stream = sourceTopic.split("/")[1];
+                StreamMessage streamMessage = new StreamMessage(stream, preparePayload(
+                        SMBridge.APPEND_TIME_DEFAULT_STREAM, SMBridge.APPEND_TOPIC_DEFAULT_STREAM, message));
+                try {
+                    smClient.publish(streamMessage);
+                    LOGGER.atInfo().kv("Source Topic", message.getTopic()).kv("Destination Stream", stream)
+                            .log("Published message");
+                } catch (SMClientException e) {
```

Do you want to do retry? 

Also by swallowing the exception, does it mean message getting lost?

---

Not sure what you mean "message getting lost." The error message from the exception will be logged. The MQTT message **will** be lost at the moment. 
Retry would be better in SMClient rather than message bridge. Happy to discuss, but I think it's probably too late to add for this scope. Especially because it would likely include validation of customer-configured parameters, since publish fails likely happen as a result of SM rejecting certain append or other function call.

---

Added todo